ruby/lib/minitest/queue/runner.rb

```diff
@@ -76,76 +80,141 @@ def report_command
+            It's automatically inferred on Buildkite, CircleCI and Travis.
+          EOS
+          opts.separator ""
+          opts.on('--build BUILD_ID', *help) do |build_id|
+            queue_config.build_id = build_id
           end
 
-          opts.on('--seed SEED') do |seed|
-            queue_config.seed = seed
+          help = split_heredoc(<<-EOS)
+            Optional. Sets a prefix for the build id in case a single CI build runs multiple independent test suites.
+              Example: --namespace integration
+          EOS
+          opts.separator ""
+          opts.on('--namespace NAMESPACE', *help) do |namespace|
+            queue_config.namespace = namespace
           end
 
-          opts.on('--timeout TIMEOUT') do |timeout|
+          opts.separator ""
+          opts.separator "COMMANDS"
+          opts.separator ""
+          opts.separator "    run [TEST_FILES...]: Participate in leader election, and then work off the test queue."
+
+          help = split_heredoc(<<-EOS)
+            Specify a timeout after which if a test haven't completed, it will be picked up by another worker.
+            It is very important to set this vlaue higher than the slowest test in the suite, otherwise performance will be impacted.
+            Defaults to 30 seconds.
+          EOS
+          opts.separator ""
+          opts.on('--timeout TIMEOUT', *help) do |timeout|
             queue_config.timeout = Float(timeout)
           end
 
-          opts.on('--namespace NAMESPACE') do |namespace|
-            queue_config.namespace = namespace
+          help = split_heredoc(<<-EOS)
+            Specify $LOAD_PATH directory, similar to Ruby's -I
+          EOS
+          opts.separator ""
+          opts.on('-IPATHS', *help) do |paths|
+            self.load_paths = paths
           end
 
-          opts.on('--build BUILD_ID') do |build_id|
-            queue_config.build_id = build_id
+          help = split_heredoc(<<-EOS)
+            Sepcify a seed used to shuffle the test suite.
+            On Buildkite, CircleCI and Travis, the commit revision will be used by default.
+          EOS
+          opts.separator ""
+          opts.on('--seed SEED', *help) do |seed|
+            queue_config.seed = seed
           end
 
-          opts.on('--worker WORKER_ID') do |worker_id|
+          help = split_heredoc(<<-EOS)
+            A unique identifier for this worker, It must be consistent to allow retries.
+            If not specified, retries won't be available.
+            It's automatically inferred on Buildkite and CircleCI.
+          EOS
+          opts.separator ""
+          opts.on('--worker WORKER_ID', *help) do |worker_id|
             queue_config.worker_id = worker_id
           end
 
+          help = split_heredoc(<<-EOS)
+            Defines how many time a single test can be requeued.
+            Defaults to 0.
+          EOS
```

Comment:
Not sure about the minimum Ruby version you had in mind, but if you'd use `<<~EOS` there's no need for stripping the string, e.g. `opts.on('--requeue-tolerance RATIO', *help.lines)` without `split_heredoc`.

Reply:
Lines have to be passed as an array of strings, otherwise the formatting is messed up. So it doesn't change much.

Also it would have been two small of an improvement for requiring 2.3. That's why I didn't use `<<~`